### DISCUSSION TITLE
Unrealistic locations when using crawlWrap function

### DISCUSSION THREAD

**Discussion by caiosantosnt** [bmcclintock/momentuHMM#131] [Q&A] [answered]
I am working with GPS data at a fine scale. The datasets have some big gaps and the location error is not negligible. 
When I interpolate missing gaps using crawlWrap I get some locations at unrealistic places, for example, land for marine animals. This can have a significant impact on my analysis because the resistance of the landscape is the main covariate. I do have a "barrier" raster file and I'm able to constrain movement (e.g. from land) when using simData but not when interpolating. 
Is there any way to deal with this issue using momentuHMM without introducing NAs to the dataset? 

I know it is sort of a basic question but I could not find any resource to help me with this.

Thanks.
Cheers

**Comment by bmcclintock** [comment, marked as answer]
It's very difficult to reliably interpolate complex movement with both measurement error and big gaps. Common two-stage approaches are: 1) break up the tracks into segments without big gaps, interpolate the segments (possibly using multiple imputation), and fit the HMM to the interpolated segments; and 2) interpolate the complete tracks (possibly using multiple imputation) and fit the HMM to the interpolated tracks (with the big gaps set to NA if the interpolated tracks are deemed unreliable during these periods). Inference in the second stage (HMM fitting) critically depends on the underlying movement model used for interpolation in the first stage (`crawl` uses a continuous-time correlated random walk). If the continuous-time random walk is not capturing key features of the movement (e.g. land avoidance), then one option would be to use the new [pathroutr](https://jmlondon.github.io/pathroutr/index.html) package to route the interpolated tracks around land barriers (possibly using multiple imputation) and then fit the HMM to the interpolated, re-routed tracks. 

Otherwise, single-stage approaches could be used to fit the HMM while simultaneously accounting for measurement error and missing data, but these approaches typically require custom (Bayesian MCMC) code and can be very computationally expensive (e.g. [McClintock et al. 2012](https://doi.org/10.1890/11-0326.1), [Michelot & Blackwell 2019](https://doi.org/10.1111/2041-210X.13154)).

**Reply by caiosantosnt** [reply]
Thanks, Brett.
It definitely gives us guidance on how to deal with this issue.
